File: byeongmin/정렬/나이순정렬.java
package sorting;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.StringTokenizer;

public class 나이순정렬 {
    static class Person implements Comparable<Person> {
        private int age;
        private String name;

        public Person(int age, String name) {
            this.age = age;
            this.name = name;
        }

        public int getAge() {
            return age;
        }

        public String getName() {
            return name;
        }

        @Override
        public int compareTo(Person o) {
            if (this.age < o.age)
                return -1;
            else if (this.age > o.age)
                return 1;
            else
                return 0;
        }
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        ArrayList<Person> list = new ArrayList<>();

        int n = Integer.parseInt(st.nextToken());

        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            int age = Integer.parseInt(st.nextToken());
            String name = st.nextToken();

            list.add(new Person(age, name));
        }

        Collections.sort(list);

        for (int i = 0; i < list.size(); i++) {
            Person person = list.get(i);
            System.out.println(person.getAge() + " " + person.getName());
        }
    }
}

File: wooyoung/백준_정렬 문제 풀기/나이순 정렬.py
# 첫째 줄에 온라인 저지 회원의 수 N이 주어진다
N = int(input())

# 둘째 줄부터 N개의 줄에는 각 회원의 나이와 이름이 공백으로 구분되어 주어진다.

member = []


for _ in range(N):
    age, name = input().split()
    member.append((int(age),name))


# 각 회원의 나이와 이름이 공백으로 구분되어 주어진다.
#age,name = input().split()

# 나이는 1보다 크거나 같으며, 200보다 작거나 같은 정수이고
# 이름은 알파벳 대소문자로 이루어져 있고, 길이가 100보다 작거나 같은 문자열이다.

#member.append(int(age),name)

# 입력은 가입한 순서로 주어진다.
# 나이순으로 정렬

member.sort(key=lambda x:x[0])
for i in member:
    print(i[0], i[1])